zigbeelumi/integrationpluginzigbeelumi.cpp

```diff
@@ -320,19 +323,98 @@ void IntegrationPluginZigbeeLumi::setupThing(ThingSetupInfo *info)
             if (cluster->clusterId() == ZigbeeClusterLibrary::ClusterIdDoorLock) {
-                // Note: shoehow the vibration sensor is using the door lock cluster, with undocumented attribitues.
-                // This device is completly out of spec, so we just recognize the vibration trough tests and it looks like
-                // attribute id 85 is the indicator for vibration. The payload contains an unsigned int, but not sure what it indicates yet
+                // Note: the vibration sensor is using the door lock cluster, with undocumented attribitues.
+                // The data interpretation has been stolen from: https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/fromZigbee.js
+                bool valueOk = false;
+                switch (attribute.id()) {
+                case 0x0055: { // motion type
+                    quint16 value = attribute.dataType().toUInt16(&valueOk);
+                    if (!valueOk) {
+                        qCWarning(dcZigbeeLumi()) << thing << "failed to convert attribute data to uint16." << attribute;
+                        return;
+                    }
 
-                if (attribute.id() == 85) {
-                    bool valueOk = false;
+                    switch (value) {
+                    case 0x01:
+                        qCDebug(dcZigbeeLumi()) << thing << "vibration detected";
+                        emit emitEvent(Event(lumiVibrationSensorVibrationDetectedEventTypeId, thing->id()));
+                        break;
+                    case 0x02:
+                        qCDebug(dcZigbeeLumi()) << thing << "tilt detected";
+                        emit emitEvent(Event(lumiVibrationSensorTiltDetectedEventTypeId, thing->id()));
+                        break;
+                    case 0x03:
+                        qCDebug(dcZigbeeLumi()) << thing << "drop detected";
+                        emit emitEvent(Event(lumiVibrationSensorDropDetectedEventTypeId, thing->id()));
+                        break;
+                    default:
+                        qCDebug(dcZigbeeLumi()) << thing << "unhandled movement type" << value;
+                        break;
+                    }
+
+                    break;
+                }
+                case 0x0503: { // angle
                     quint16 value = attribute.dataType().toUInt16(&valueOk);
                     if (!valueOk) {
                         qCWarning(dcZigbeeLumi()) << thing << "failed to convert attribute data to uint16." << attribute;
-                    } else {
-                        qCDebug(dcZigbeeLumi()) << thing << "vibration attribute changed" << value;
-                        emitEvent(Event(lumiVibrationSensorVibrationDetectedEventTypeId, thing->id()));
+                        return;
+                    }
+
+                    // Not sure which angle this is...maybe from the tilte
+                    qCDebug(dcZigbeeLumi()) << thing << "angle" << value;
```

Comment:
?

Reply:
I cannot figure out what angle this should be...or how it could be used...so just printing it, but not using it, maybe we figure out if the debug informs...

Reply:
ok. works for me.